src/components/sidebar/components.test-pw.tsx

```diff
@@ -0,0 +1,216 @@
+    <Box height="2000px" position="relative">
+      <Box
+        data-element="test-box"
+        height="100px"
+        id="bottom-box"
+        position="absolute"
+        bottom="0px"
+      >
+        I should not be scrolled into view
+      </Box>
+      <Sidebar open onCancel={() => {}}>
+        <Textbox label="textbox" />
+      </Sidebar>
+    </Box>
+  );
+};
+
+export const SidebarBackgroundScrollWithOtherFocusableContainers = () => {
+  const toast1Ref = useRef(null);
+  const toast2Ref = useRef(null);
+  return (
+    <Box height="2000px" position="relative">
+      <Box
+        data-element="test-box"
+        height="100px"
+        id="bottom-box"
+        position="absolute"
+        bottom="0px"
+      >
+        I should not be scrolled into view
+      </Box>
+      <Sidebar
+        open
+        onCancel={() => {}}
+        focusableContainers={[toast1Ref, toast2Ref]}
+      >
+        <Textbox label="textbox" />
+      </Sidebar>
+      <Toast open onDismiss={() => {}} ref={toast1Ref} targetPortalId="stacked">
+        Toast message 1
+      </Toast>
+      <Toast open onDismiss={() => {}} ref={toast2Ref} targetPortalId="stacked">
+        Toast message 2
+      </Toast>
+    </Box>
+  );
+};
+
+export const SidebarComponentFocusable = (props: Partial<SidebarProps>) => {
+  const [setIsDialogOpen] = React.useState(false);
+  const [isToastOpen, setIsToastOpen] = React.useState(false);
+  const toastRef = React.useRef(null);
+  const CUSTOM_SELECTOR = "button, .focusable-container input";
+  return (
+    <>
+      <Sidebar
+        open
+        onCancel={() => setIsDialogOpen}
+        header={<Typography variant="h3">Sidebar header</Typography>}
+        focusableContainers={[toastRef]}
+        focusableSelectors={CUSTOM_SELECTOR}
+        {...props}
+      >
+        <Box className="focusable-container">
+          <Textbox label="First Name" />
+        </Box>
+        <Box>
+          <Textbox label="Surname" />
+        </Box>
+        <Box className="focusable-container">
+          <Button
+            buttonType="primary"
+            data-element="open-toast"
+            onClick={() => setIsToastOpen(true)}
+          >
+            Show toast
+          </Button>
+        </Box>
+      </Sidebar>
+      <Toast
+        open={isToastOpen}
+        onDismiss={() => setIsToastOpen(false)}
+        ref={toastRef}
+        targetPortalId="stacked"
+        data-element="toast"
+      >
+        Toast Message
+      </Toast>
+    </>
+  );
+};
+
+export const TopModalOverride = () => {
+  const [isOpen1, setIsOpen1] = useState(true);
+  const [isOpen2, setIsOpen2] = useState(true);
+  return (
+    <>
+      <Sidebar
+        open={isOpen1}
+        onCancel={() => setIsOpen1(false)}
+        header="Sidebar"
+        topModalOverride
+      >
+        foo
+      </Sidebar>
+      <Dialog open={isOpen2} onCancel={() => setIsOpen2(false)} title="Dialog">
+        foo
+      </Dialog>
+    </>
+  );
+};
```

Comment:
We can then delete them from `sidebar-test.stories.tsx`, since these mock examples are moved to `components.test-pw.tsx`.